src/main/scala/swiss/dasch/domain/MimeTypeGuesser.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2021 - 2023 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package swiss.dasch.domain
+
+import eu.timepit.refined.types.string.NonEmptyString
+import swiss.dasch.domain.PathOps.fileExtension
+import zio.nio.file.Path
+import zio.{IO, ZIO, ZLayer}
+
+import java.io.IOException
+
+final case class MimeType private (value: NonEmptyString) extends AnyVal {
+  def stringValue: String = value.value
+}
+object MimeType {
+
+  def unsafeFrom(str: String): MimeType =
+    from(str).fold(msg => throw new IllegalArgumentException(msg), identity)
+
+  def from(str: String): Either[String, MimeType] =
+    Option(str)
+      .toRight("Mime type cannot be null")
+      .flatMap(it => NonEmptyString.from(it).left.map(_ => "Mime type cannot be empty"))
+      .map(MimeType.apply)
+}
+
+final case class MimeTypeGuesser() {
+
+  private val allMappings: Map[String, MimeType] = SupportedFileType.values.flatMap(_.mappings).toMap
+
+  def guess(file: Path): IO[IOException, Option[MimeType]] =
+    ZIO.succeed(allMappings.get(file.fileExtension.toLowerCase))
```

Comment:
in what scenario would this be `ZIO.fail(IOException)`? Wouldn't it make more sense to make this of type `Option[MimeType]` directly (or if you want it to be a ZIO, make it `ZIO.fromOption`)?

Reply:
I initially designed the signature for a guesser which would be looking into the contents of a file rather than checking the file extension only. In that case an `IOException` might happen. I left the signature so that later we will still have the option to do a more thorough file content check of the file with something like [Apache Tika](https://tika.apache.org/).

But I guess you are right YAGNI.

Reply:
It doesn't hurt either, if you want to leave it in

Reply:
See abf318958def7a285af4413dc32d24383c304771